### PR TITLE
Add implementations for mix_up

### PR DESCRIPTION
**Pull request by shashaka**
I have implemented MixUp image processing. If this implementation is acceptable, I can proceed to add the corresponding test cases.

here is my [gist](https://colab.research.google.com/drive/1lpFMvyqN5pFNxdtAjr8pOMpJS_kezkvm?usp=sharing)

### CODE REVIEW

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    labels = tf.cast(tf.one_hot(labels, 10), tf.float32)
+    mixup = keras.layers.preprocessing.MixUp(10)
+    augmented_images, updated_labels = mixup(
+        {'images': images, 'labels': labels}
+    )
+    # output == {'images': updated_images, 'labels': updated_labels}
+    ```
+    """
+
+    def __init__(self, alpha=0.2, seed=None, **kwargs):
```
Comment (fchollet) [inline keras/src/layers/preprocessing/image_preprocessing/mix_up.py:37]: Please add support for the `data_format` argument (defaulting to None, and which could be `"channels_last"` or `"channels_first"`)

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        super().__init__(**kwargs)
+        self.alpha = alpha
+        self.seed = seed
+        self.generator = SeedGenerator(seed)
+
+    def _sample_from_beta(self, alpha, beta, shape, seed):
```
Comment (fchollet) [inline keras/src/layers/preprocessing/image_preprocessing/mix_up.py:43]: There is a `keras.random.beta` op that you might be able to use.

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            batch_size = self.backend.shape(images)[0]
+
+        if seed is None:
+            seed = self._get_seed_generator(self.backend._backend)
+
+        permutation_order = self.backend.random.shuffle(
+            self.backend.numpy.arange(0, batch_size), seed=seed
+        )
+
+        lambda_sample = self._sample_from_beta(
```
Comment (fchollet) [inline keras/src/layers/preprocessing/image_preprocessing/mix_up.py:76]: Why lambda?

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+)
+from keras.src.random import SeedGenerator
+
+
+@keras_export("keras.layers.MixUp")
+class MixUp(BaseImagePreprocessingLayer):
+    """MixUp implements the MixUp data augmentation technique.
+
+    Args:
+        alpha: Float between 0 and 1. Inverse scale parameter for the gamma
```
Comment (fchollet) [inline keras/src/layers/preprocessing/image_preprocessing/mix_up.py:13]: Typically you'd want to control intensity via `factor` parameter, where 0 would mean "no mixup" and 1 would mean "maximum mixup". This is much easier for users to understand.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py (hunk 1 above)
Comment (shashaka) [reply keras/src/layers/preprocessing/image_preprocessing/mix_up.py]: I added data_format argument.

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py (hunk 2 above)
Comment (shashaka) [reply keras/src/layers/preprocessing/image_preprocessing/mix_up.py]: I replaced this method to keras.random.beta

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py (hunk 3 above)
Comment (shashaka) [reply keras/src/layers/preprocessing/image_preprocessing/mix_up.py]: I renamed the variable to mix_weight for better clarity and ease of understanding.

File: keras/src/layers/preprocessing/image_preprocessing/mix_up.py (hunk 4 above)
Comment (shashaka) [reply keras/src/layers/preprocessing/image_preprocessing/mix_up.py]: I updated the description based on your guidance.

**Comment by shashaka** [comment]
I have updated the source code as your guidance. plz let me know any corrections are needed.

File: keras/src/layers/preprocessing/image_preprocessing/mix_up_test.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import numpy as np
+import pytest
+
+from keras.src import layers
+from keras.src import testing
+
+
+class MixUpTest(testing.TestCase):
```
Comment (fchollet) [inline keras/src/layers/preprocessing/image_preprocessing/mix_up_test.py:8]: Please add a test for tf.data compatibility (like what we have for the other augmentation layers)

Comment (shashaka) [reply keras/src/layers/preprocessing/image_preprocessing/mix_up_test.py:9]: I've added tf_compatiblity test.

**Comment by shashaka** [comment]
I have added additional test cases. If further corrections are needed, please let me know.

**Review by fchollet — APPROVED** [review]
Excellent -- thank you for the contribution!
